app/assets/javascripts/notebook/magic/customC3Chart.coffee

```diff
@@ -32,17 +21,15 @@ define([
     console.log("chartOptions", chartOptions)
     chartOptions.bindto = "#" + chart_container.attr("id")
     chartOptions.data = {} if not chartOptions.data
-    chartOptions.data.rows = []
+    chartOptions.data.json = @dataInit
 
-    chart = c3.generate(chartOptions)
+    chartOptions.data.keys = { value: options.headers } if not chartOptions.data.keys or not chartOptions.data.keys.value
 
-    chart.load({ rows: prepare_c3_data(options.headers, @dataInit) })
-    chart.flush()
+    chart = c3.generate(chartOptions)
 
     dataO.subscribe( (newData) =>
-      chart.unload();
-      chart.load({
-        rows: prepare_c3_data(options.headers, newData)
-      });
+      chartOptions.data.json = newData#_.filter(newData, (obj) -> !_.findWhere(currentData, obj))
+      #chartOptions.data.unload = true
```

Comment:
if `new` data is superset of the old (i.e. `immutable/deterministic` additions) - duplicates are likely fine (didn't check).
but in case of random sampling this is not true.